src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -1250,12 +1273,18 @@ trait Contexts { self: Analyzer =>
       }
     }
 
-  } //class Context
+    def isPackageOwnedInDifferentUnit(s: Symbol): Boolean =
+      if (s.isOverloaded) s.alternatives.exists(isPackageOwnedInDifferentUnit)
```

Comment:
How can a package-owned symbol be overloaded?

Reply:
package object?

Reply:
Good point! Hadn't considered those... FTR, had to convince myself in the REPL:

```
scala> val s = typeOf[scala.Error].typeSymbolDirect // type alias defined in the scala package object
s: $r.intp.global.Symbol = type Error

scala> s.isDefinedInPackage
res24: Boolean = true

scala> s.effectiveOwner
res26: $r.intp.global.Symbol = package scala

scala> s.effectiveOwner.hasPackageFlag
res27: Boolean = true

scala> s.owner
res28: $r.intp.global.Symbol = package object scala

scala> s.owner.hasPackageFlag
res29: Boolean = false
```

Reply:
TFTRR! (thanks for the repl record) I was about to try it as well, but distracted with backporting the "long result printing fix" for hrhino. The REPL looks so different way back in 2.12!

Reply:
I also have to forward port a few niceties, inevitably.